From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: gstelementfactory: don't use uri handler code without uri handler support

When configuring gstreamer with --disable-uri, libgstreamer-0.10.so
breaks:

../gst/.libs/libgstreamer-0.10.so: undefined reference to `gst_uri_handler_get_type'
collect2: ld returned 1 exit status
make[3]: *** [gst-inspect-0.10] Error 1
make[3]: Leaving directory `/home/rsc/svn/oselas/bsp/pengutronix/OSELAS.BSP-Pengutronix-AllYes-trunk/platform-phyCORE-PXA270/build-target/gstreamer-0.10.20/tools'

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 gst/gstelementfactory.c |    2 ++
 1 file changed, 2 insertions(+)

Index: gstreamer-0.10.20/gst/gstelementfactory.c
===================================================================
--- gstreamer-0.10.20.orig/gst/gstelementfactory.c
+++ gstreamer-0.10.20/gst/gstelementfactory.c
@@ -305,6 +305,7 @@ gst_element_register (GstPlugin * plugin
   factory->numpadtemplates = klass->numpadtemplates;
   klass->elementfactory = factory;
 
+#ifndef GST_DISABLE_URI
   /* special stuff for URI handling */
   if (g_type_is_a (type, GST_TYPE_URI_HANDLER)) {
     GstURIHandlerInterface *iface = (GstURIHandlerInterface *)
@@ -326,6 +327,7 @@ gst_element_register (GstPlugin * plugin
     if (!factory->uri_protocols)
       goto urierror;
   }
+#endif
 
   interfaces = g_type_interfaces (type, &n_interfaces);
   for (i = 0; i < n_interfaces; i++) {
